### DISCUSSION TITLE
This seems to be a bug but I can't tell.

### DISCUSSION THREAD

**Discussion by carsonwright** [adonisjs/core#2096] [General]
```
export default class User extends BaseModel {
  @hasMany(() => Answer, {
    foreignKey: 'userID',
  })
  
  public answers: HasMany<typeof Answer>
  }
 ```

Why is it forcing a user_id relationship...?
``` 
[1] error: select * from "answers" where "user_id" = $1 limit $2 - column "user_id" does not exist
```

**Comment by HappyLDE** [comment]
It is forcing user_id because you specifically told it so by foreignKey: 'userId'. I don't see anything wrong here. Answers should have a user_id field.

**Reply by carsonwright** [reply]
I want to use userID in the database, the system should not rewrite my tables. I told it to user userID, I want it to user userID. How do I get the system to just use my desired capitalization?

**Reply by carsonwright** [reply]
The answer should have a "userID" field. Does adonis not support this? I would have preferred to know I couldn't use camel case columns and tables from the beginning.

**Reply by carsonwright** [reply]
I have switched to using underscores, but just don't love it. I'll add a serializer to render the json to correct formatting at the end.

**Comment by RomainLanz** [comment]
Hey @carsonwright! 👋 

Could you please share the code that create this query?

Also, takes note that AdonisJS expect you to have snake_cased fields inside your database (as per SQL convention), but it also transform them to camelCase inside your code (as per JS convention).
